sda/cmd/api/api.go

```diff
@@ -344,6 +354,205 @@ func deleteFile(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+// reencryptHeader re-encrypts the header of a file using the public key
+// provided in the request header and returns the new header. The function uses
+// gRPC to communicate with the re-encrypt service and handles TLS configuration
+// if needed. The function also handles the case where the CA certificate is
+// provided for secure communication.
+func reencryptHeader(oldHeader []byte, reEncKey string) ([]byte, error) {
```

Comment:
This function should have a separate test.

Reply:
Added in commit ea308ac7